### PR TITLE
Dtls13: ack management improvements

### PR DESCRIPTION
**Pull request by rizlik**
Improve DTLS 1.3 ACK record handling, enforcing record count limits. It also introduces comprehensive tests for these edge cases.

* Added a strict upper bound (`DTLS13_ACK_MAX_RECORDS`) on the number of ACK records that can be tracked and encoded

### CODE REVIEW

**Comment by rizlik** [comment]
retest this please

**Comment by rizlik** [comment]
retest this please

**Comment by rizlik** [comment]
retest this please

**Comment by programsurf** [comment]
Verified this fixes the vulnerability I reported.

- `Dtls13GetAckListLength()` (word16 truncation) fully removed
- Three independent guards: insert-time cap (128), write-time validation, in-loop bounds check
- Counter stays consistent across insert / flush / remove / write-dup transfer
- Mutex early-return fixes are a nice bonus

Original PoC (4097 records -> 65KB heap overflow) no longer reproduces. LGTM.

— Sunwoo Lee and Seunghyun Yoon, Korea Institute of Energy Technology (KENTECH)

File: src/dtls13.c [hunk 1]
```diff
@@ -2731,6 +2750,7 @@ int Dtls13DoScheduledWork(WOLFSSL* ssl)
                 tail = &(*tail)->next;
             *tail = ssl->dtls13Rtx.seenRecords;
             ssl->dtls13Rtx.seenRecords = NULL;
+            ssl->dtls13Rtx.seenRecordsCount = 0;
             ssl->dupWrite->sendAcks = 1;
```
[the remark this answers is not in the document]
Comment (rizlik) [reply src/dtls13.c:2754]: Not introduced in this PR, won't fix here.

**Comment by rizlik** [comment]
retest this please
